src/main/kotlin/graphics/scenery/proteins/RibbonDiagram.kt

```diff
@@ -446,7 +446,10 @@ class RibbonDiagram(val protein: Protein, private val displaySS: Boolean = false
             for (i in 0..count) {
                 guidePointsWithoutDummy[i].ssLength++
             }
-            val dummyVecBeg = caBegin.randomFromVector()
+            // spline calculation needs a beginning point. In this case, we simply take the vector from the second to
+            // the first point and add it to the first curve point. Thus, we create a new starting point of the curve
+            val dummyVecBeg = Vector3f(caBegin).add(Vector3f(caBegin).min(Vector3f(aminoList[1].getAtom("CA").getVector()))
```

Comment:
scenery has Kotlin extension functions for operators like `+`/`add` on `Vector*f` - please use those instead of the written out form, it's quite hard to comprehend otherwise.